src/Workspaces/Core/Portable/Editing/ImportAdderService.cs

```diff
@@ -53,7 +53,7 @@ internal abstract class ImportAdderService : ILanguageService
             var existingNamespaces = new HashSet<INamespaceSymbol>();
             await this.GetExistingImportedNamespacesAsync(document, model, existingNamespaces, cancellationToken).ConfigureAwait(false);
 
-            var namespacesToAdd = new HashSet<INamespaceSymbol>(namespaces);
+            var namespacesToAdd = new HashSet<INamespaceSymbol>(namespaces, NamespaceEqualityComparer.Singleton);
```

Comment:
I have tried using .Equals. When debugging, it calls into Symbol.Equals, which does reference equality:

```csharp
        // By default, we do reference equality. This can be overridden.
        public override bool Equals(object obj)
        {
            return (object)this == obj;
        }

        public bool Equals(ISymbol other)
        {
            return this.Equals((object)other);
        }
```

I chose this approach to localize the fix, without affecting lots of working code.